src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -383,100 +397,281 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
+                {
+                    newParameters[separatorIndex + 1] = newParameters[separatorIndex + 1].WithPrependedLeadingTrivia(separator.TrailingTrivia);
+                    separator = separator.WithoutTrailingTrivia();
+                }
 
-                newParameters.Add(param);
+                newSeparators[separatorIndex] = separator;
             }
 
-            var numSeparatorsToSkip = originalParameters.Count - reorderedParameters.Count;
-            return SyntaxFactory.SeparatedList(newParameters, GetSeparators(list, numSeparatorsToSkip));
+            return SyntaxFactory.SeparatedList(newParameters, newSeparators);
         }
 
-        private static T TransferLeadingWhitespaceTrivia<T>(T newArgument, SyntaxNode oldArgument) where T : SyntaxNode
+        private static (T newParamNode, SyntaxToken newSeparator) TransferTrivia<T>(
+            SeparatedSyntaxList<T> list,
+            T newParamNode,
+            int originalIndex,
+            int newIndex,
+            int reorderedParametersCount) where T : SyntaxNode
         {
-            var oldTrivia = oldArgument.GetLeadingTrivia();
-            var oldOnlyHasWhitespaceTrivia = oldTrivia.All(t => t.IsKind(SyntaxKind.WhitespaceTrivia));
+            // Copy whitespace trivia from original position.
+            newParamNode = TransferLeadingWhitespaceTrivia(newParamNode, list[newIndex]);
+
+            // The separator associated with the node should have its trivia transferred over as well. If we're looking at the last node, then there is no associated separator.
+            var originalSeparator = originalIndex < list.Count - 1 ? list.GetSeparator(originalIndex) : default;
+            var newSeparator = newIndex < reorderedParametersCount - 1 ? list.GetSeparator(newIndex) : default;
+
+            // If the associated node is not switching positions, we don't need to do any work.
+            if (originalSeparator == newSeparator)
+            {
+                return (newParamNode, newSeparator);
+            }
+
+            // If we have a comment of format '//', we switch the trivia notation to /* */.
+            newParamNode = ChangeNodeSingleLineCommentsToMultiLine(newParamNode);
+            originalSeparator = ChangeSeparatorSingleLineCommentsToMultiLine(originalSeparator);
+
+            // Case 1: The current node in the updated configuration is the last parameter.
+            // Append any trailing trivia from the original separator to the trailing trivia of the new node, as long as it's not all whitespace. We also want to ignore end-of-line trivia.
+            if (newSeparator.IsKind(SyntaxKind.None))
+            {
+                if (originalSeparator.TrailingTrivia.Any(trivia => !trivia.IsKind(SyntaxKind.WhitespaceTrivia)))
+                {
+                    newParamNode = newParamNode.WithAppendedTrailingTrivia(originalSeparator.TrailingTrivia.Where(trivia => !trivia.IsKind(SyntaxKind.EndOfLineTrivia)));
```

Comment:
so i don't actually understand the first two bits.  say we have whitespace followed by newline.  it looks like we then concat the whitespace, but not hte newline.  i don't understand why this is desirable.  needs example in code.